waspc/data/Generator/templates/sdk/wasp/client/router/index.ts

```diff
@@ -4,25 +4,22 @@ import type {
   RouteDefinitionsToRoutes,
   OptionalRouteOptions,
   ParamValue,
+  ExpandRouteOnOptionalStaticSegments,
 } from './types'
 
 // PUBLIC API
 export const routes = {
   {=# routes =}
   {= name =}: {
     to: "{= urlPath =}",
-    {=#  hasUrlParams =}
     build: (
-      options: {
+      options{=^ hasUrlParams =}?{=/ hasUrlParams =}:
+      OptionalRouteOptions
+      {=#  hasUrlParams =}& {
         params: {{=# urlParams =}{= name =}{=# isOptional =}?{=/ isOptional =}: ParamValue;{=/ urlParams =}}
-      } & OptionalRouteOptions,
-    ) => interpolatePath("{= urlPath =}", options.params, options?.search, options?.hash),
-    {=/ hasUrlParams =}
-    {=^ hasUrlParams =}
-    build: (
-      options?: OptionalRouteOptions,
-    ) => interpolatePath("{= urlPath =}", undefined, options?.search, options?.hash),
-    {=/ hasUrlParams =}
+      }{=/ hasUrlParams =}
+      {=# hasOptionalStaticSegments =}& { path: ExpandRouteOnOptionalStaticSegments<"{= urlPath =}"> }{=/ hasOptionalStaticSegments =}
```

Comment:
If a route has optional static segments e.g. `/my/:id/route?` which can be either `/my/:id` or `/my/:id/route`, we require the user to specify which variant they want to use by providing the `path` argument.

Reply:
I agree 100% 👍 

I realised we have 2 degrees of freedom (has params, has optional static segments) and didn't want to do 4 variants. But 2 variants are enough, makes it a lot cleaner!